### PR TITLE
Removing some cruff and updating signatures for consistency

### PR DESCRIPTION
**Pull request by drisspg**
# Summary

- This removes a bunch of example score mods that were primarily used for testing and places them directly in the test file. We should follow up with merging test_flex_decode and test_flash when the velocity slows down a little
- Fixes a bug with indexing on block mask
- Adds some doc strings to helper funcs and fixes some misc typing things
- Forces functions passed to `create_block_mask` to mask_mods and updates tests files

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131013
* #131012
* #130904
* __->__ #130871

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/nn/attention/flex_attention.py [hunk 1]
```diff
@@ -302,20 +301,34 @@ def __getitem__(self, index) -> "BlockMask":
             mask_mod=self.mask_mod,
         )
 
+    def __repr__(self):
```
Comment (drisspg) [inline torch/nn/attention/flex_attention.py:304]: I found this helpful for debugging but its a big repr

File: torch/nn/attention/flex_attention.py [hunk 2]
```diff
@@ -295,27 +294,51 @@ def __str__(self):
     def __getitem__(self, index) -> "BlockMask":
         new_kv_num_blocks = self.kv_num_blocks[index]
         new_kv_indices = self.kv_indices[index]
+        new_kv_num_blocks_full = (
```
Comment (drisspg) [inline torch/nn/attention/flex_attention.py:297]: cc @Chillee  I think this a bug from lower in the stack

Comment (Chillee) [reply torch/nn/attention/flex_attention.py:297]: mm.... yes, this is a bug. Do you want me to fix it lower down in the stack?

Comment (drisspg) [reply torch/nn/attention/flex_attention.py:297]: whatever makes it easiest to land lol

File: torch/nn/attention/flex_attention.py [hunk 3]
```diff
@@ -544,12 +561,22 @@ def create_mask(
             raise AssertionError
 
 
-# Done as a workaround around torch.compile not compiling what we want in the
-# presence of the torchfunctionmdoe
 def _create_block_mask_inner(
-    mod_fn, B, H, M, N, device, KV_BLOCK_SIZE, Q_BLOCK_SIZE, mod_type
+    mod_fn,
```
Comment (Chillee) [inline torch/nn/attention/flex_attention.py:565]: Can rename this to `mask_mod`

**Review by joydddd — APPROVED** [review]

**Review by Chillee — APPROVED** [review]

File: torch/nn/attention/flex_attention.py (hunk 3 above)
Comment (drisspg) [reply torch/nn/attention/flex_attention.py:565]: will do

**Pull request closed without merging**